File: src/main/java/com/architecture_map/belarus/service/impl/ArchitectServiceImpl.java
package com.architecture_map.belarus.service.impl;

import com.architecture_map.belarus.dto.ArchitectDto;
import com.architecture_map.belarus.entity.Architect;
import com.architecture_map.belarus.mapper.ArchitectMapper;
import com.architecture_map.belarus.repository.ArchitectRepository;
import com.architecture_map.belarus.service.ArchitectService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class ArchitectServiceImpl implements ArchitectService {

    private final ArchitectRepository architectRepository;
    private final ArchitectMapper architectMapper;

    @Override
    public Architect create(ArchitectDto architectDto) {
        return architectRepository.save(architectMapper.mappingToArchitect(architectDto));
    }

    @Override
    public Optional<Architect> findById(Integer id) {
        return architectRepository.findById(id);
    }

    @Override
    public List<Architect> findAll() {
        return architectRepository.findAll();
    }
}

File: src/main/kotlin/by/architecturemap/belarus/service/impl/ArchitectServiceImpl.kt
package by.architecturemap.belarus.service.impl

import by.architecturemap.belarus.entity.Architect
import by.architecturemap.belarus.exception.NotFoundException
import by.architecturemap.belarus.repository.jpa.ArchitectRepository
import by.architecturemap.belarus.service.ArchitectService
import org.springframework.stereotype.Service

@Service
class ArchitectServiceImpl(
    private val architectRepository: ArchitectRepository
) : ArchitectService {

    override fun create(architect: Architect): Architect = architectRepository.save(architect)

    override fun find(id: Int): Architect = architectRepository.findById(id)
        .orElseThrow { throw NotFoundException("Architect not found with id: $id") }

    override fun findAll(): List<Architect> = architectRepository.findAll()
}
